bindings/java/src/main/com/apple/foundationdb/FutureKey.java

```diff
@@ -22,15 +22,23 @@
 
 import java.util.concurrent.Executor;
 
+import com.apple.foundationdb.EventKeeper.Events;
+
 class FutureKey extends NativeFuture<byte[]> {
-	FutureKey(long cPtr, Executor executor) {
+	private final EventKeeper eventKeeper;
+	FutureKey(long cPtr, Executor executor, EventKeeper eventKeeper) {
 		super(cPtr);
 		registerMarshalCallback(executor);
+		this.eventKeeper = eventKeeper;
 	}
 
 	@Override
 	protected byte[] getIfDone_internal(long cPtr) throws FDBException {
-		return FutureKey_get(cPtr);
+		byte[] bytes = FutureKey_get(cPtr);
+		if (eventKeeper != null) {
+			eventKeeper.count(Events.BYTES_FETCHED, bytes.length);
```

Comment:
I'm not sure if there's a great solution to this, but I believe this function is called with the `pointerReadLock` held.

Reply:
🤦  you're right! In this case, it's probably _relatively_ mild, since that logic is executed in its own thread, but still not desirable.

I don't know if this solution is great either, but I moved the logic to a `thenApply()` function on the Future itself, which should move this outside the lock without a ton of bad consequences.